zcash_client_memory/src/proto/memory_wallet.proto

```diff
@@ -0,0 +1,186 @@
+  repeated TreeEndHeightsRecord orchard_tree_shard_end_heights = 14;
+
+  // UTXOs known to this wallet
+  repeated TransparentReceivedOutputRecord transparent_received_outputs = 15;
+  // UTXOs spent by this wallet
+  repeated TransparentReceivedOutputSpendRecord transparent_received_output_spends = 16;
+  // Map from spends to their location in the blockchain
+  repeated TransparentSpendCacheRecord transparent_spend_map = 17;
+  // Queue of transaction data requests the wallet should make to the lightwalletd provided to obtain more complete information
+  repeated TransactionDataRequest transaction_data_requests = 18;
+}
+
+message Accounts {
+  repeated Account accounts = 1; // map from account index to account data
+  uint32 account_nonce = 2; // the nonce for the next account
+}
+
+message Account {
+  uint32 account_id = 1; // the index of this account
+  AccountKind kind = 2; // derived or imported
+
+  optional bytes seed_fingerprint = 3;
+  optional uint32 account_index = 4; // HD index to derive account from seed
+
+  optional AccountPurpose purpose = 5; // spending or view-only
+
+  string viewing_key = 6; // the viewing key for this account
+  AccountBirthday birthday = 7; // the block height at which this account was created
+  repeated Address addresses = 8; // account addresses
+
+  repeated EphemeralAddressRecord ephemeral_addresses = 9; // map from index to encoded unified address
+  string account_name = 10; // human readable name for the account
+  optional string key_source = 11; // key source metadata
+}
+
+enum AccountKind {
+  Derived = 0;
+  Imported = 1;
+}
+
+enum AccountPurpose {
+  Spending = 0;
+  ViewOnly = 1;
+}
+
+message AccountBirthday {
+  ChainState prior_chain_state = 1; // the chain state at the block height before the account was created
+  optional uint32 recover_until = 2; // the block height until which the account should stop being in recovery mode
+}
+
+// A record storing transaction data in the transaction table
+message TransactionTableRecord {
+  TxId tx_id = 1;
+  TransactionEntry tx_entry = 2;
+}
+
+// Maps a block height and transaction index to a transaction ID.
+message TxLocatorRecord {
+  uint32 block_height = 1;
+  uint32 tx_index = 2;
+  TxId tx_id = 3;
+}
+
+message EphemeralAddress {
+  string address = 1;
+  optional bytes used_in_tx = 2;
+  optional bytes seen_in_tx = 3;
+}
+
+message EphemeralAddressRecord {
+  uint32 index = 1;
+  EphemeralAddress ephemeral_address = 2;
+}
+
+message ChainState {
+  uint32 block_height = 1; // the height of this block
+  bytes block_hash = 2;
+  bytes final_sapling_tree = 3;
+  bytes final_orchard_tree = 4;
+}
+
+message WalletBlock {
+  uint32 height = 1; // the height of this block
+  bytes hash = 2; // the ID (hash) of this block, same as in block explorers
+  uint32 block_time = 3; // Unix epoch time when the block was mined
+  repeated bytes transactions = 4; // the txids of transactions in this block
```

Comment:
This is the protobuf field corresponding to my previous comment. There is only implicit mention that these might only be wallet-relevant transactions, and there is no documentation that this is a set (and thus the protobuf array order must be strictly ignored).